### PR TITLE
Routing for merge

### PR DESCRIPTION
**Pull request by Tamila-Bodnar**

### CODE REVIEW

**Comment by Tamila-Bodnar** [comment]
fix conflicts

File: src/app/core/services/authentication.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthenticationService {
+  TOKEN_KEY = 'token_key';
+  constructor(private router: Router) { }
+
+  whatRole(): string{
```
Comment (rmurtazin) [inline src/app/core/services/authentication.service.ts:11]: add access modificators all over

File: src/app/core/services/authentication.service.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return 'admin';
+  }
+
+  saveToken(token: string): void{
+    window.localStorage.removeItem(this.TOKEN_KEY);
```
Comment (rmurtazin) [inline src/app/core/services/authentication.service.ts:16]: you can use without window

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -2,7 +2,22 @@ import { NgModule } from '@angular/core';
...
+import { HomeComponent } from './shared/components/home/home.component';
+
+import { AuthGuard } from './core/guards/auth.guard';
+
+const routes: Routes = [
+  { path: '', redirectTo: 'login', pathMatch: 'full' },
+  { path: 'login'},
+  { path: 'home',
+    canActivate: [AuthGuard],
+    children : [
```
[the remark this answers is not in the document]
Comment (Tamila-Bodnar) [reply src/app/app-routing.module.ts:14]: Yes, it's a part of the home page and I added like a child of home in line 16. Did I add in the wrong way?

File: src/app/core/services/authentication.service.ts (hunk 2 above)
Comment (Tamila-Bodnar) [reply src/app/core/services/authentication.service.ts]: Thanks

File: src/app/app-routing.module.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tamila-Bodnar) [reply src/app/app-routing.module.ts:13]: Thanks

File: src/app/core/guards/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class AuthGuard implements CanActivate {
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if (this.authenticationService.checkAuthentication()) {
+        return true;
+      }
+      this.router.navigate(['']);
+      return false;
```
Comment (rmurtazin) [inline src/app/core/guards/auth.guard.ts:18]: in if condition you need to check availability of token in localStorage and if it absent there navigate to ['login'] and in return you should invoke a method to get status by token from back-end. For now after invoking this method  add || true.

File: src/app/core/guards/auth.guard.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if (!this.loginService.getToken()) {
+        this.router.navigate(['login']);
+        return true;
+      }
+      if (this.loginService.getUser() || true ) {
+        return true;
+      }
+
+      return false;
```
[the remark this answers is not in the document]
Comment (Tamila-Bodnar) [reply src/app/core/guards/auth.guard.ts:22]: I tried. I then have a type mismatch

**Review by rmurtazin — APPROVED** [review]

**Pull request merged**